src/Sparrow/Json/BlittableJsonReaderObject.cs

```diff
@@ -23,18 +23,77 @@ public sealed unsafe class BlittableJsonReaderObject : BlittableJsonReaderBase,
         private byte* _metadataPtr;
         private readonly int _size;
         private readonly int _propCount;
-        private readonly long _currentOffsetSize;
-        private readonly long _currentPropertyIdSize;
+
         private readonly bool _isRoot;
         private readonly byte* _objStart;
         private readonly byte* _propNames;
+
+        private readonly int _currentOffsetSize;
+        private readonly int _currentPropertyIdSize;
         private readonly int _propNamesDataOffsetSize;
 
+        private readonly int _functionTableIndex;
+
         public DynamicJsonValue Modifications;
 
         private Dictionary<StringSegment, object> _objectsPathCache;
         private Dictionary<int, object> _objectsPathCacheByIndex;
 
+
+
+
+        static BlittableJsonReaderObject()
+        {
+            // PERF: We will create a function table that ensures that it has enough space to handle the mapping. 
+            FunctionTable = new FunctionTableItem[0b01_000000];
```

Comment:
Because we are doing this with bits, so I wanted to make it clear that we are using 6 bits only. The table is not really using all of the items, there are many nulls in there.